src/textual/widgets/_input.py

```diff
@@ -221,6 +221,7 @@ def __init__(
         *,
         suggester: Suggester | None = None,
         validators: Validator | Iterable[Validator] | None = None,
+        prevent_validation_on: Iterable[type[Message]] | None = None,
```

Comment:
This parameter is negative condition. The dev is saying don't do something on this. This creates a bit of cognitive load where the developer has to reverse what they want in their mind.

Suggest we drop the `prevent_` and `have it simply `validate_on`.

Requiring an iterable of messages could also be confounding. As the typing would accept any Message.

I think this should be an iterable of literals. Something like this:

`validate_on: Iterable[ValidateCondition] | None`

Where 

`ValidateContition = Literal["changed", "submitted", "blur"]`

Reply:
Re `Literal`: consider it done.

I disagree with the negative condition & cognitive load argument.
One negation is something reasonable.
Since the default is “everything”, it is sensible to have a parameter that allows ignoring just some, just like with the context manager `prevent`.

Reply:
There is only three options. `validate_on = {"blur", "changed"}` tells me everything I need to know. `prevent_validation_on = {"changed"}` leaves me wondering what the default is.

Reply:
See a26307278